cvat/apps/tf_annotation/views.py

```diff
@@ -18,329 +19,370 @@
-    def _normalize_box(box, w, h):
-        xmin = int(box[1] * w)
-        ymin = int(box[0] * h)
-        xmax = int(box[3] * w)
-        ymax = int(box[2] * h)
-        return xmin, ymin, xmax, ymax
-
-    result = {}
-    model_path = os.environ.get('TF_ANNOTATION_MODEL_PATH')
-    if model_path is None:
-        raise OSError('Model path env not found in the system.')
-    job = rq.get_current_job()
-
-    detection_graph = tf.Graph()
-    with detection_graph.as_default():
-        od_graph_def = tf.GraphDef()
-        with tf.gfile.GFile(model_path + '.pb', 'rb') as fid:
-            serialized_graph = fid.read()
-            od_graph_def.ParseFromString(serialized_graph)
-            tf.import_graph_def(od_graph_def, name='')
-
-        try:
-            config = tf.ConfigProto()
-            config.gpu_options.allow_growth=True
-            sess = tf.Session(graph=detection_graph, config=config)
-            frames = frame_provider.get_frames(frame_provider.Quality.ORIGINAL)
-            for image_num, (image, _) in enumerate(frames):
-
-                job.refresh()
-                if 'cancel' in job.meta:
-                    del job.meta['cancel']
-                    job.save()
-                    return None
-                job.meta['progress'] = image_num * 100 / len(frame_provider)
-                job.save_meta()
-
-                image = Image.open(image)
-                width, height = image.size
-                if width > 1920 or height > 1080:
-                    image = image.resize((width // 2, height // 2), Image.ANTIALIAS)
-                image_np = load_image_into_numpy(image)
-                image_np_expanded = np.expand_dims(image_np, axis=0)
-
-                image_tensor = detection_graph.get_tensor_by_name('image_tensor:0')
-                boxes = detection_graph.get_tensor_by_name('detection_boxes:0')
-                scores = detection_graph.get_tensor_by_name('detection_scores:0')
-                classes = detection_graph.get_tensor_by_name('detection_classes:0')
-                num_detections = detection_graph.get_tensor_by_name('num_detections:0')
-                (boxes, scores, classes, num_detections) = sess.run([boxes, scores, classes, num_detections], feed_dict={image_tensor: image_np_expanded})
-
-                for i in range(len(classes[0])):
-                    if classes[0][i] in labels_mapping.keys():
-                        if scores[0][i] >= treshold:
-                            xmin, ymin, xmax, ymax = _normalize_box(boxes[0][i], width, height)
-                            label = labels_mapping[classes[0][i]]
-                            if label not in result:
-                                result[label] = []
-                            result[label].append([image_num, xmin, ymin, xmax, ymax])
-        finally:
-            sess.close()
-            del sess
-    return result
+	(im_width, im_height) = image.size
```

Comment:
It seems you use tab as indentation, please revert.

Reply:
Ok, will do.

Reply:
Updated file.